From: "Mike Rapoport (IBM)" <rppt@kernel.org>
Subject: loongarch: drop ranges for definition of ARCH_FORCE_MAX_ORDER
Date: Wed, 22 Mar 2023 10:17:27 +0200

LoongArch defines insane ranges for ARCH_FORCE_MAX_ORDER allowing
MAX_ORDER up to 63, which implies maximal contiguous allocation size of
2^63 pages.

Drop bogus definitions of ranges for ARCH_FORCE_MAX_ORDER and leave it a
simple integer with sensible defaults.

Users that *really* need to change the value of ARCH_FORCE_MAX_ORDER will
be able to do so but they won't be mislead by the bogus ranges.

Link: https://lkml.kernel.org/r/20230322081727.2516291-1-rppt@kernel.org
Signed-off-by: Mike Rapoport (IBM) <rppt@kernel.org>
Acked-by: Kirill A. Shutemov <kirill.shutemov@linux.intel.com>
Reviewed-by: David Hildenbrand <david@redhat.com>
Cc: Huacai Chen <chenhuacai@kernel.org>
Cc: WANG Xuerui <kernel@xen0n.name>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 arch/loongarch/Kconfig |    3 ---
 1 file changed, 3 deletions(-)

--- a/arch/loongarch/Kconfig~loongarch-drop-ranges-for-definition-of-arch_force_max_order
+++ a/arch/loongarch/Kconfig
@@ -420,11 +420,8 @@ config NODES_SHIFT
 
 config ARCH_FORCE_MAX_ORDER
 	int "Maximum zone order"
-	range 13 63 if PAGE_SIZE_64KB
 	default "13" if PAGE_SIZE_64KB
-	range 11 63 if PAGE_SIZE_16KB
 	default "11" if PAGE_SIZE_16KB
-	range 10 63
 	default "10"
 	help
 	  The kernel memory allocator divides physically contiguous memory
_
